tests/waku_archive/test_driver_postgres.nim

```diff
@@ -24,23 +26,29 @@ proc computeTestCursor(pubsubTopic: PubsubTopic,
   )
 
 suite "Postgres driver":
+  ## Unique driver instance
+  var driver {.threadvar.}: ArchiveDriver
 
-  const storeMessageDbUrl = "postgres://postgres:test123@localhost:5432/postgres"
+  asyncSetup:
+    let driverRes = await newTestPostgresDriver()
+    if driverRes.isErr():
+      assert false, driverRes.error
 
-  asyncTest "Asynchronous queries":
-    let driverRes = PostgresDriver.new(dbUrl = storeMessageDbUrl,
-                                       maxConnections = 100)
+    driver = driverRes.get()
 
-    assert driverRes.isOk(), driverRes.error
+  asyncTeardown:
+    let resetRes = await cast[PostgresDriver](driver).reset()
+    if resetRes.isErr():
+      assert false, resetRes.error
 
-    let driver = driverRes.value
-    discard await driver.reset()
+    (await driver.close()).expect("driver to close")
 
+  asyncTest "Asynchronous queries":
     var futures = newSeq[Future[ArchiveDriverResult[void]]](0)
 
     let beforeSleep = now()
     for _ in 1 .. 100:
-      futures.add(driver.sleep(1))
+      futures.add(cast[PostgresDriver](driver).sleep(1))
```

Comment:
This case was needed because `driver` was declared as `ArchiveDriver`. We implement a simple inheritance hierarchy where all driver implementations inherit from `ArchiveDriver`. In this particular case, the only driver implementation that has `sleep` implemented is the `PostgresDriver`. Enhanced in https://github.com/waku-org/nwaku/pull/2477/commits/4b8d76e8502d51c38bb118e52ecc7466167b4547